example/push_notifications/lib/main.dart

```diff
@@ -3,26 +3,33 @@ import 'package:flutter/material.dart';
 import 'package:talkjs_flutter/talkjs_flutter.dart';
 import 'package:firebase_core/firebase_core.dart';
 import 'firebase_options.dart';
+import 'package:flutter_apns_only/flutter_apns_only.dart';
+import 'package:flutter_local_notifications/flutter_local_notifications.dart';
 
 Future<void> main() async {
   WidgetsFlutterBinding.ensureInitialized();
 
+  // Request push notification permissions
   if (Platform.isAndroid) {
-    await Firebase.initializeApp(
-      options: DefaultFirebaseOptions.currentPlatform,
-    );
+    await FlutterLocalNotificationsPlugin()
+        .resolvePlatformSpecificImplementation<
+            AndroidFlutterLocalNotificationsPlugin>()!
+        .requestNotificationsPermission();
+  } else if (Platform.isIOS) {
+    await ApnsPushConnectorOnly().requestNotificationPermissions(
+        const IosNotificationSettings(sound: true, alert: true, badge: true));
```

Comment:
This exposes the implementation details that we're using the `flutter_local_notifications` and `flutter_apns_only` plugins internally.
I don't like it, because then we can't easily switch to a different implementation (for example the Dart/Flutter wrapper on the native TalkJS package that we wanted to do).
I propose to add some methods to the `Talk` object that will request the permissions, so that we can change internal implementation whenever we want.

Reply:
> This exposes the implementation details that we're using the flutter_local_notifications and flutter_apns_only plugins internally.

I used those since it would simplify the example.

> because then we can't easily switch to a different implementation

We can. If a user relies on a transitive dependency, it comes with the inherent risk that the dependency may go at some point.

> I propose to add some methods to the Talk object that will request the permissions, so that we can change internal implementation whenever we want.

Mhh.. I was initially against this idea because I felt that we should let our users handle this on their own however, now that I think of it, we could provide such a method.